airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -67,11 +77,8 @@ public AirbyteConnectionStatus check(final JsonNode config) {
           .setUseLegacySql(false)
           .build();
 
-      // GCS upload time re-uses destination-GCS for check and other uploading (CSV format writer)
       if (UploadingMethod.GCS.equals(uploadingMethod)) {
-        final GcsDestination gcsDestination = new GcsDestination();
-        final JsonNode gcsJsonNodeConfig = BigQueryUtils.getGcsJsonNodeConfig(config);
-        final AirbyteConnectionStatus airbyteConnectionStatus = gcsDestination.check(gcsJsonNodeConfig);
+        final AirbyteConnectionStatus airbyteConnectionStatus = checkStorageIamPermissions(config);
```

Comment:
```suggestion
        // TODO: use GcsDestination::check instead of writing our own custom logic to check perms
        // this is not currently possible because using the Storage class to check perms requires
        // a service account key, and the GCS destination does not accept a Service Account Key, 
        // only an HMAC key
        final AirbyteConnectionStatus airbyteConnectionStatus = checkStorageIamPermissions(config);
```